### PR TITLE
Em 3582 hrs create logging formater

### PR DESCRIPTION
**Pull request by crimesp**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-3582

### Change description ###
Implement Log formatting  as per CCD style with non db logger
added sharee auditing
general refactoring of audit service
added client Ip and ccd case id to audit logs

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/AuditEntryService.java [hunk 1]
```diff
@@ -1,83 +1,127 @@
...
+
     @Autowired
     private SecurityService securityService;
 
-    public List<HearingRecordingAuditEntry> findHearingRecordingAudits(HearingRecording hearingRecording) {
+    @Autowired
+    private AuditLogFormatter auditLogFormatter;
+
+    void logOnly(Long caseId, AuditActions action) {
+        var entry = new LogOnlyAuditEntry();
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/AuditEntryService.java:45]: var?

File: src/main/java/uk/gov/hmcts/reform/em/hrs/controller/HearingRecordingController.java [hunk 2]
```diff
@@ -145,24 +149,17 @@ public ResponseEntity getSegmentBinary(@PathVariable("recordingId") UUID recordi
...
-
-        Map<String, String> segmentDetails = downloadService.getDownloadInfo(recordingId, segmentNo);
-        String filename = segmentDetails.get("filename");
-
-        response.setHeader(HttpHeaders.CONTENT_TYPE, segmentDetails.get("contentType"));
-        response.setHeader(
-            HttpHeaders.CONTENT_LENGTH,
-            segmentDetails.get("contentLength")
-        );//will be overriden if partial request
-
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/controller/HearingRecordingController.java:159]: so is this not required anymore? download was failing when we weren't setting these headers

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java [hunk 3]
```diff
@@ -43,41 +41,32 @@ public SegmentDownloadServiceImpl(HearingRecordingSegmentRepository segmentRepos
...
-
-        LOGGER.info(
-            "downloading blob with the following properties: [filenmae: {}, content-type: {}, content-length: {}]",
-            segment.getFilename(), blobProperties.getContentType(), blobProperties.getBlobSize()
-        );
-
-        return Map.of(
-            "filename", segment.getFilename(),
-            "contentType", blobProperties.getContentType(),
-            "contentLength", String.valueOf(blobProperties.getBlobSize())
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java:65]: do we not need to set these details anymore?

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java [hunk 4]
```diff
@@ -116,17 +105,14 @@ public void download(String filename, HttpServletRequest request,
                     request.getHeader(HttpHeaders.CONTENT_LENGTH)
```
Comment (crimesp) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java:105]: content length here

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java [hunk 5]
```diff
@@ -43,48 +42,43 @@ public SegmentDownloadServiceImpl(HearingRecordingSegmentRepository segmentRepos
...
+            .logHttpHeaders(request);//keep during early life support to assist with any range or other issues.
 
         String rangeHeader = HttpHeaderProcessor.getHttpHeaderByCaseSensitiveAndLowerCase(request, HttpHeaders.RANGE);
         LOGGER.info("Range header for filename {} = {}", filename, rangeHeader);
 
         BlobRange blobRange = null;
-        if (rangeHeader != null) {
-            long fileSize = blobstoreClient.getFileSize(filename);
+        if (rangeHeader == null) {
+            response.setHeader(HttpHeaders.CONTENT_LENGTH, String.valueOf(fileSize));
```
Comment (crimesp) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java:80]: and here

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java [hunk 6]
```diff
@@ -43,48 +42,43 @@ public SegmentDownloadServiceImpl(HearingRecordingSegmentRepository segmentRepos
...
+        auditEntryService.createAndSaveEntry(segment, AuditActions.USER_DOWNLOAD_REQUESTED);
+
+        String filename = segment.getFilename();
+        BlobInfo blobInfo = blobstoreClient.fetchBlobInfo(filename);
+        long fileSize = blobInfo.getFileSize();
+        String contentType = blobInfo.getContentType();
+        String attachmentFilename = String.format("attachment; filename=%s", filename);
+
+        response.setHeader(HttpHeaders.CONTENT_DISPOSITION, attachmentFilename);
+        response.setHeader(HttpHeaders.CONTENT_TYPE, contentType);
```
Comment (crimesp) [inline src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java:68]: content type here

File: src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java (hunk 3 above)
Comment (crimesp) [reply src/main/java/uk/gov/hmcts/reform/em/hrs/service/SegmentDownloadServiceImpl.java:65]: have moved - please see comments below as to where

**Review by benouaer — APPROVED** [review]

**Pull request merged**